### PR TITLE
DRILL-7094: UnSupported Bson type: DECIMAL128 Mongodb 3.6 onwards

### PR DESCRIPTION
**Pull request by rajeshguptaiiitm**
Handling Decimal128 by converting into Double.

# [DRILL-7094](https://issues.apache.org/jira/browse/DRILL-7094): UnSupported Bson type: DECIMAL128 Mongodb 3.6 onwards

(UnSupported Bson type: DECIMAL128 Mongodb 3.6 onwards)

## Description

Handling the Decimal128 by converting into Double.

## Documentation
Now Bson Decimal128 type will be handled for the SQL queries and Show the result in double value.

## Testing
Created a Document in Mongo DB with field of Decimal128 type and executing a SQL query using Apache Drill and it return the results. Previously there was a error unsupported Bson Type.

### CODE REVIEW

**Comment by cgivre** [comment]
@rajeshguptaiiitm 
Thank you for the contribution.  Could you add a unit test for this fix?
Thanks!

**Comment by cgivre** [comment]
LGTM +1

Thanks for the contribution!

**Review by cgivre — APPROVED** [review]
LGTM +1

**Comment by rajeshguptaiiitm** [comment]
Hi,
Could you please tell me how can I merge these changes because merge option is not available for me.

**Comment by cgivre** [comment]
@rajeshguptaiiitm 
You have to be a committer to merge changes.  Please squash all commits and set the commit message to 
```
DRILL-7094: UnSupported Bson type: DECIMAL128 Mongodb 3.6 onwards
```
I'll merge once that's done.  Thanks!

**Comment by rajeshguptaiiitm** [comment]
Have done it. Kindly merge it.

**Comment by cgivre** [comment]
@rajeshguptaiiitm 
The commits are not squashed.   There should be only 1 commit.  
The way to do this is:

```
git reset --soft HEAD~4
git commit -m "DRILL-7094: UnSupported Bson type: DECIMAL128 Mongodb 3.6 onwards"
git push origin +master
```
Please verify your branch BEFORE executing the last step.

**Comment by rajeshguptaiiitm** [comment]
Now I have done the described steps.

**Comment by cgivre** [comment]
@rajeshguptaiiitm 
I'm not quite sure where the breakdown was.  To merge this into Drill, I need one commit with all the changes, with the message that has the PR number and description.   

The commands I gave you above should have done exactly what I described.  However, we now have 6 commits instead of 1.  Can you please take another look and try?
Thanks!

**Comment by rajeshguptaiiitm** [comment]
I have done the steps again.

**Comment by cgivre** [comment]
@rajeshguptaiiitm 
I'm not sure what you did, but now there are 59 files modified.   Can you please revert this and try again?  The goal is to have one single commit with your specific changes (and only those changes).  This commit now includes a bunch of commits that are not related to this PR. 
Just to be clear this command `git reset --soft HEAD~4` the `4` refers to the number of commits you are squashing, so that can change.

**Comment by cgivre** [comment]
@rajeshguptaiiitm 
I'm not sure what you're doing at this point, but I'd recommend starting a new branch with just the files you've modified.  I'm not a git expert, so that's the best advice I can give at this point.

**Comment by cgivre** [comment]
@rajeshguptaiiitm 
I'm not sure what you did but we're not getting the results we're looking for.  The commit that will get merged must have:

* A single commit with the code you modified and nothing else
* A commit message with the pull request number and description

Once we're there, I am happy to merge this.

**Comment by rajeshguptaiiitm** [comment]
Now it is all set to go

**Comment by cgivre** [comment]
@rajeshguptaiiitm 
Thanks for your work on this.  I know working with git can be challenging.   We're almost there.  Unfortunately, there are still some extra files in the most recent commit.  Can you please make sure that the commit only has the changes from this PR and we'll get it merged.
Thanks!

**Pull request closed without merging**
